source/mir/range.d

```diff
@@ -0,0 +1,92 @@
+/++
+Ranges.
+
+See_also: $(MREF mir,_primitives).
+
+License:   $(HTTP boost.org/LICENSE_1_0.txt, Boost License 1.0).
+Copyright: Copyright © 2017-, Ilya Yaroshenko
+Authors:   Ilya Yaroshenko
++/
+module mir.range;
+
+/++
+Data size counter.
+
+Does not store anything
++/
+struct Counter(T)
+{
+    import std.range: isInputRange, ElementType;
+    import std.traits: isImplicitlyConvertible, isSomeChar;
+    ///
+    size_t _count;
```

Comment:
Why do you expose this?

Reply:
To make hacks easily. This looks useful.